File: old_backend/important-core-code/java/VulnerablePath.java
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical path traversal vulnerabilities for testing Aikido scanner

import java.io.*;
import javax.servlet.http.*;

public class VulnerablePath extends HttpServlet {
    
    public void doGet(HttpServletRequest request, HttpServletResponse response) {
        String filename = request.getParameter("file");
        
        try {
            // VULNERABLE: Direct file access
            File file = new File("/var/data/" + filename);
            FileInputStream fis = new FileInputStream(file);
            
            // VULNERABLE: Path concatenation
            String path = request.getParameter("path");
            BufferedReader reader = new BufferedReader(
                new FileReader(path)
            );
            
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: old_backend/important-core-code/php/vulnerable-path.php
<?php
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical path traversal vulnerabilities for testing Aikido scanner

// VULNERABLE: Direct file inclusion
$page = $_GET['page'];
include("/var/www/pages/" . $page);

// VULNERABLE: File reading
$filename = $_GET['file'];
$content = file_get_contents($filename);
echo $content;

// VULNERABLE: readfile with user input
$document = $_REQUEST['doc'];
readfile("/documents/" . $document);

// VULNERABLE: require with concatenation
$module = $_GET['module'];
require($module . '.php');
?>
